### ISSUE TITLE
Adding gql property to type user

### ISSUE THREAD

**Issue by auBusinessDaD** [cleverbeagle/pup#352]
Hey there,
I am trying to add a new property to the user type query in \ui\queires\users.gql and every time I change the property in there it break the who users gql functions.
I have added the property everywhere I can find - startup, api and ui, an when I remove the property from the \ui\queires\users.gql file everything is working AND I can use the use the playground to see the property on users profiles. But for some reason I get a 400 error whenever I start up the instance with that property in \ui\queires\users.gql file...?
Would love some help please.

**Comment by auBusinessDaD** [comment]
I had thought it was something to do with the Apollo or cache, but even when I spin up brand new implementations it still doesn't like it.

**Comment by cleverbeagle** [comment]
When do you see the 400 error? Automatically on server startup or only when making requests? If only when making requests, double-check the arguments for the query match in your schema as well as on the client (e.g., is an arg required in the schema but not in the client).

Something like:

```
// Schema

type Query {
  someQuery(requiredArg: String!): Stuff
}

// Client

query someQuery($requiredArg: String) { // Notice that String is not marked as required here.
  someQuery(requiredArg: $requiredArg) {
    field
  }
}
```

Also worth making sure that in your startup/client/apollo.js file, in the `onError` callback, you're logging out all of the errors (sometimes an error gets burried here that's helpful for debugging). For example:

```
onError(({ graphQLErrors, networkError }) => {
    if (graphQLErrors) {
    graphQLErrors.map(({ message, locations, path }) => {
      console.log(message);
    });
    }

    if (networkError) {
      console.log(networkError);
    }
}),
```

**Comment by auBusinessDaD** [comment]
Get no notifications in the terminal that there is an issue. Only get the issue when I go to a page that uses the call - so it doesn't load and the console shows the following error;
![image](https://user-images.githubusercontent.com/22830607/108288195-4a69e900-71e0-11eb-8813-74c9da2a5073.png)

```
----from api\Users\types.js
type Friends {
    userId: String
    username: String
  }

type User {
    _id: String
    name: Name
    username: String
    emailAddress: String
    oAuthProvider: String
    roles: [Role]
    itchlist: [String]
    wishlist: [String]
    ownlist: [String]
    friends: [Friends]
    subscription: String
    subscriptionId: String
    gamePlayCounts: [GamePlayCount],
    loanedTo: [LoanedTo]
    borrowedFrom: [BorrowedFrom]
    settings: [UserSetting] # From /api/UserSettings/types.js
  }

----from ui\queries\Users.gql----
query userFromUsername($username: String) {
  userFromUsername(username: $username) {
    ...UserAttributes,
      username
      wishlist
      itchlist
      ownlist
      subscription
      friends {
        userId
        username
      }
      loanedTo {
        _id
        userId
        username
      }
      borrowedFrom {
        _id
        userId
        username
      }
      gamePlayCounts {
        _id
        count
      }
  }
}
```

As for Apollo, that does have an error link - but I don't see anything in the console whent he error occurs (just in the browser console/dev tools)

```
const errorLink = onError(({ graphQLErrors, networkError }) => {
  if (graphQLErrors)
    graphQLErrors.map(({ message, location, path }) =>
      console.log(`[GraphQL error]: Message: ${message}, Location: ${location}, Path: ${path}`),
    );

  if (networkError) console.log(`[Network error]: ${networkError}`);
});
```

**Comment by auBusinessDaD** [comment]
OH... MY... GOH!!

Turns out, I am just an idiot. The issue was not with my code. Everything was right - and I could see that in the above. BUT, and wait for the face palm, because I am using docker images sitting side by side for the real site and the dev site I change some settings manually AND this is where the rub is. YES, I changed the code correctly and even got everything looking sexy (even though I couldn't yet see it) BUT the dev site was still looking at the live site containers gql (instead of the dev gql).

I will put some s=time into automating that deployment to docker... I am loving Docker to manage it all, but manually handling the dev to live push (by manually changing 2 values is a risk of my stupidity seeping in....)
